### PR TITLE
Connects Falkonry/tracker#814 Facts API Change

### PR DESCRIPTION
**Pull request by dilpeshjain**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -733,70 +733,135 @@ from falkonryclient import schemas as Schemas
...
+assessmentId = 'id of the assessment'
 assessmentResponse = fclient.get_assessment(assessmentId)
 
 // aprioriConditionList 
 conditionList = assessment.get_aprioriConditionList()
 
 ```
 
-#### Add facts data (json format) to Assessment
+#### Add facts data (json format) to Assessment of a multi entity datastream
```
Comment (aniket-amrutkar) [inline README.md]: • Add one test case for using the tagIdentifier
• Add one test case for sending the additional tag while uploading facts

File: falkonryclient/service/falkonry.py [hunk 2]
```diff
@@ -195,7 +253,67 @@ def add_facts_stream(self, assessment, data_type, options, data):
         :param options: dict
         :param data: Stream
         """
-        url = '/assessment/' + assessment + '/facts'
+
+        firstReqParam = True
+
+        url = '/assessment/' + assessment + '/facts?'
+
+        if 'startTimeIdentifier' in options:
```
Comment (aniket-amrutkar) [inline falkonryclient/service/falkonry.py]: Please create a function for generating url. It is same in both add_facts, add_facts_stream

File: test/TestAddFacts.py [hunk 3]
```diff
@@ -96,8 +108,19 @@ def test_add_csv_facts(self):
...
             try:
                 resp_assessment = fclient.create_assessment(asmtRequest)
-                data = "time,end,car,Health\n2011-03-31T00:00:00Z,2011-04-01T00:00:00Z,IL9753,Normal\n2011-03-31T00:00:00Z,2011-04-01T00:00:00Z,HI3821,Normal"
-                response = fclient.add_facts(resp_assessment.get_id(), 'csv', {}, data)
+                data = "time,end,car,Health\n2011-03-31T00:00:00.000Z,2011-04-01T00:00:00.000Z,IL9753,Normal\n2011-03-31T00:00:00.000Z,2011-04-01T00:00:00.000Z,HI3821,Normal"
+
+                options = {
+                    'startTimeIdentifier': "time",
+                    'endTimeIdentifier': "end",
+                    # 'timeFormat': "YYYY-MM-DDTHH:mm:ss.SSSZ",
```
Comment (aniket-amrutkar) [inline test/TestAddFacts.py]: Remove commented code.
And add description to the test case that this is for single entity datastream

File: test/TestAddFactsStream.py [hunk 4]
```diff
@@ -43,9 +43,17 @@ def test_add_json_facts(self):
...
-
-                response = fclient.add_facts(resp_assessment.get_id(), 'json', {}, data)
+                data = io.open('./test/factsData.json')
+
+                options = {
+                    'startTimeIdentifier': "time",
+                    'endTimeIdentifier': "end",
+                    'timeFormat': "iso_8601",
+                    'timeZone': time.get_zone(),
+                    # 'entityIdentifier': "car",
```
Comment (aniket-amrutkar) [inline test/TestAddFactsStream.py]: Remove commented code.
And add description to the test case that this is for single entity datastream

**Comment by phagunbaya** [comment]
@dilpeshjain update ?

File: test/TestAddFactsStream.py [hunk 5]
```diff
@@ -94,8 +104,18 @@ def test_add_csv_facts(self):
 
             try:
                 resp_assessment = fclient.create_assessment(asmtRequest)
-                data = io.open('./factsData.csv')
-                response = fclient.add_facts(resp_assessment.get_id(), 'csv', {}, data)
+                data = io.open('./test/factsData.csv')
```
Comment (aniket-amrutkar) [inline test/TestAddFactsStream.py]: Why did you change the path?
